figures/monolith-id-generation-class.mmd

```diff
@@ -0,0 +1,49 @@
+classDiagram
+    class BalancerManager {
+        - balancerConnections: BalancerConnection[]
+        - bus: EventEmitter
+        + addBalancerConnection(conn: BalancerConnection): void
+        + getConnection(id: string): BalancerConnection | undefined
+        - onBalancerConnect(conn: BalancerConnection): void
+        - onBalancerDisconnect(conn: BalancerConnection, code: number, reason: string): void
+        - onBalancerMessage(conn: BalancerConnection, message: MsgB2M): void
+        - onBalancerError(conn: BalancerConnection, error: WebSocket.ErrorEvent): void
+        + on(event: BalancerManagerEvents, handler: BalancerManagerEventHandlers): void
+        - emit(event: BalancerManagerEvents, ...args: any[]): void
+    }
+
+    class BalancerConnection {
+        <<abstract>>
+        - id: string
+        - bus: EventEmitter
+        + send(message: MsgM2B): Result<void, Error>
+        + disconnect(): Result<void, Error>
+        - emit(event: BalancerConnectionEvents, ...args: any[]): void
+        + on(event: BalancerConnectionEvents, handler: BalancerConnectionEventHandlers): void
```

Comment:
feel free to abbreviate some of these function signatures to make the diagram readable